### PR TITLE
Added the ability to deploy to a normal kubernetes cluster.

### PR DESCRIPTION
**Pull request by pajikos**
This pull request is not primarily intended for approval at this time, but as a basis for discussion to introduce compatibility with the normal Kubernetes cluster.

### Related Issue
No related issue

### Description of changes
- Updated certDefaultMode value from 384 to 416
- Changed oc command to a dynamic CLI_EXEC variable
- Fixed operator-sdk command timeout to 5 minutes
- Updated kustomize version to v4.5.7

#### Changes moved to PR #122 for additional discussion
- ~~Updated POSTGRES_IMAGE value from registry.redhat.io/rhel8/postgresql-13:1-56 to quay.io/fedora/postgresql-13~~ 
- ~~Added support for generating certificates in Kubernetes clusters~~

### CODE REVIEW

File: scripts/deploy.sh [hunk 1]
```diff
@@ -28,9 +28,9 @@ uninstaller() {
     echo "* Deleting operator"
     eval "$UNINSTALL_COMMAND"
     echo "* Deleting quay.io secret"
-    oc delete secret $EXPECTED_SECRET_NAME -n "$INSTALL_NAMESPACE" || true
+    $CLI_EXEC delete secret $EXPECTED_SECRET_NAME -n "$INSTALL_NAMESPACE" || true
     echo "* Reenable search v1"
-    oc patch mch "${mch_name}" -n "$INSTALL_NAMESPACE" --type=merge -p '{"spec":{"overrides":{"components":[{"name":"search","enabled": true}]}}}'
+    # oc patch mch "${mch_name}" -n "$INSTALL_NAMESPACE" --type=merge -p '{"spec":{"overrides":{"components":[{"name":"search","enabled": true}]}}}'
```
Comment (jlpadilla) [inline scripts/deploy.sh:33]:
```suggestion
    if [ "$cluster_type" == "OpenShift" ]; then
      oc patch mch "${mch_name}" -n "$INSTALL_NAMESPACE" --type=merge -p '{"spec":{"overrides":{"components":[{"name":"search","enabled": true}]}}}'
    fi
```

**Comment by jlpadilla** [comment]
@pajikos I suggest we split this PR so we can merge some changes now.

### Changes that can be merged now, won't affect existing CI pipeline.
- Updated certDefaultMode value from 384 to 416
- Changed oc command to a dynamic CLI_EXEC variable
- Fixed operator-sdk command timeout to 5 minutes
- Updated kustomize version to v4.5.7

### Changes that need additional discussion or updates to existing CI pipeline.
- Updated POSTGRES_IMAGE value from registry.redhat.io/rhel8/postgresql-13:1-56 to quay.io/fedora/postgresql-13
- Added support for generating certificates in Kubernetes clusters

**Comment by jlpadilla** [comment]
@pajikos This project requires signed-off commits. Please add `-s` to your commits to pass the DCO check. 
Example: `git commit -s -m "Commit description"`

**Comment by pajikos** [comment]
Hi,
some changes made:
* added signed-off to all commits (including backward)
* as suggested, a new PR (#122) created with non-compliant changes
* non-compliant changes removed from this PR
* The commit with desc: `Updated certDefaultMode value from 384 to 416` is related to the PosgreSQL image change (in #122), but I leave it here, hope no problem
* Merged suggested change made by [jlpadilla](https://github.com/jlpadilla)

**Review by jlpadilla — APPROVED** [review]
/approve
/lgtm

**Review by SherinV — APPROVED** [review]

**Pull request merged**